### ISSUE TITLE
Loss function

### ISSUE THREAD

**Issue by mehdimashayekhi** [salesforce/ebm_calibration_nlu#2]
Hi,
Thanks for sharing the code. I have a QQ, in the paper it is said that the total loss is a joint loss (equation 8), but looking at the code I can only see the NCE loss , and I don't see CE loss, maybe I missed it, and I appreciate your response. Here is the NCE loss I am referring https://github.com/salesforce/ebm_calibration_nlu/blob/e0598923551c4587e0ea8c4feb001cb9cc736103/nce_glue/my_modeling_roberta.py#L209

Thanks

**Comment by cloudygoose** [comment]
Hi thanks for your interest.
It is a joint loss.
If you look at the training iteration in my_trainer.py:
https://github.com/salesforce/ebm_calibration_nlu/blob/e0598923551c4587e0ea8c4feb001cb9cc736103/nce_glue/my_trainer.py#L679
Firstly, self._training_step will be called, which backwards the CE loss. Then nce_forward is called, which adds on the NCE loss.

**Comment by mehdimashayekhi** [comment]
Got it, thanks for your response and clarification!!
